pywbem_mock/_methodprovider.py

```diff
@@ -180,86 +180,104 @@ def __init__(self, cimrepository=None):
 
     ####################################################################
     #
-    #   Server responder for InvokeMethod.
+    #   InvokeMethod mock WBEM server responder
     #
     ####################################################################
 
     def InvokeMethod(self, namespace, MethodName, ObjectName, Params):
         # pylint: disable=invalid-name,no-self-use
         """
-        Defines the API and return for a mock WBEM server responder for
-        :meth:`~pywbem.WBEMConnection.InvokeMethod` for both static (ObjectName
-        is a class name) and dynamic (ObjectName is a CIMInstanceName) CIM
-        methods.
-
-        This method should never be called because there is no concept of a
-        default InvokeMethod in CIM.  All method providers specify specific
-        actions.
+        Implements a mock WBEM server responder for
+        :meth:`pywbem.WBEMConnection.InvokeMethod`.
+
+        Invoke a CIM method (static or dynamic) on a target CIM object (class or
+        instance) in the CIM repository of the mock WBEM server.
+
+        This default provider always raises CIMError(CIM_ERR_METHOD_NOT_FOUND)
+        because there is no concept of a default method invocation behavior in
+        in CIM (other than raising this error). A user-defined method provider
+        is necessary to have a meaningful implementation of a method invocation.
+
+        Validation already performed by the provider dispatcher that calls
+        this provider method:
+        - The Python types of all input parameters to this provider method are
+          as specified below.
+        - The namespace exists in the CIM repository.
+        - For instance-level use:
+          - The creation class of the target instance exists in the namespace
+            of the CIM repository.
+          - The target instance exists in the namespace of the CIM repository.
+          - The creation class of the target instance exposes the method to be
+            invoked.
+        - For class-level use:
+          - The target class exists in the namespace of the CIM repository.
+          - The target class exposes the method to be invoked.
+        - The set of specified CIM method input parameters matches exactly the
+          set of input parameters defined by the method in the CIM repository,
+          w.r.t. parameter name and type-related attributes (type, is_array,
+          embedded_object).
```

Comment:
Should add a not validated section to include 1) static qualifier, 2)IN OUT (required/optional) qualifiers

Reply:
We could in fact check that in the dispatcher.

Reply:
I have added corresponding code to the dispatcher.